src/Riverline/MultiPartParser/Part.php

```diff
@@ -46,7 +46,7 @@ public function __construct($content)
     protected function parseContentString($content)
     {
         // Split headers and body
-        $splits = preg_split('/(\r?\n){2}/', $content, 2);
```

Comment:
This makes things more difficult to tackle, I really wanted to avoid regexps since they will be a major performance problem when parsing large bodies or streams.

I'd suggest to detect actual separator on the stage of headers parsing (where contents is not very big typically) and use determined separator on later stages when parsing body. What you think about this approach?

My use case is parsing http requests, so having just `\n` separator there is very unlikely (to be hones I've end up writing own even more customized implementation of stream parser from scratch last few days, but anyway want to add support for streams in this library as well, so that others can use it).